ChangeSet
  1.2039 04/10/03 09:35:52 alan@lxorguk.ukuu.org.uk[torvalds] +1 -0
  [PATCH] usb: hcd locking fix
  
  Missing up() on an error path.
  
  Cc: Greg KH <greg@kroah.com>
  Signed-off-by: Andrew Morton <akpm@osdl.org>
  Signed-off-by: Linus Torvalds <torvalds@osdl.org>

  drivers/usb/core/hcd.c
    1.96 04/10/02 23:14:39 alan@lxorguk.ukuu.org.uk[torvalds] +1 -0
    usb: hcd locking fix

.........................................................................
diff -p -Nru a/drivers/usb/core/hcd.c b/drivers/usb/core/hcd.c
--- a/drivers/usb/core/hcd.c	Mon Oct  4 12:40:36 2004
+++ b/drivers/usb/core/hcd.c	Mon Oct  4 12:40:36 2004
@@ -708,6 +708,7 @@ int usb_register_bus(struct usb_bus *bus
 		bus->busnum = busnum;
 	} else {
 		printk (KERN_ERR "%s: too many buses\n", usbcore_name);
+		up(&usb_bus_list_lock);
 		return -E2BIG;
 	}
 
.........................................................................
# vim: syntax=diff
